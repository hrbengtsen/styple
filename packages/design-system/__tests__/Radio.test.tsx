import * as React from "react";
import { render, RenderResult } from "@testing-library/react";
import { Radio, RadioGroup } from "..";

describe("Test Radio, RadioGroup", () => {
  it("should render as expected", () => {
    const view: RenderResult = render(
      <RadioGroup defaultValue="1">
        <Radio value="1" />
        <Radio value="2" />
      </RadioGroup>
    );

    expect(view).toMatchSnapshot();
  });
});
