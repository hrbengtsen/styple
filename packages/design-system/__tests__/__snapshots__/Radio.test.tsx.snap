// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Radio, RadioGroup should render as expected 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        aria-required="false"
        class="c-dhzjXW"
        dir="ltr"
        role="radiogroup"
        style="outline: none;"
        tabindex="0"
      >
        <button
          aria-checked="true"
          class="c-fCoYhJ"
          data-radix-collection-item=""
          data-state="checked"
          role="radio"
          tabindex="-1"
          type="button"
          value="1"
        >
          <span
            class="c-ePBftz"
            data-state="checked"
          />
        </button>
        <button
          aria-checked="false"
          class="c-fCoYhJ"
          data-radix-collection-item=""
          data-state="unchecked"
          role="radio"
          tabindex="-1"
          type="button"
          value="2"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      aria-required="false"
      class="c-dhzjXW"
      dir="ltr"
      role="radiogroup"
      style="outline: none;"
      tabindex="0"
    >
      <button
        aria-checked="true"
        class="c-fCoYhJ"
        data-radix-collection-item=""
        data-state="checked"
        role="radio"
        tabindex="-1"
        type="button"
        value="1"
      >
        <span
          class="c-ePBftz"
          data-state="checked"
        />
      </button>
      <button
        aria-checked="false"
        class="c-fCoYhJ"
        data-radix-collection-item=""
        data-state="unchecked"
        role="radio"
        tabindex="-1"
        type="button"
        value="2"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
